File: Python/1094 - Experiments.py
n = int(input())
count = [0,0,0]
animais = ["coelhos","ratos","sapos"]
for i in range(n):
    l1 = input().split()
    if (l1[1] == "C"):
        count[0] += int(l1[0])
    elif (l1[1] == "R"):
        count[1] += int(l1[0])
    else:
        count[2] += int(l1[0])
TOTAL = (count[0]+count[1]+count[2])
print("Total: %d cobaias" % TOTAL)
for i in range(len(animais)):
    print("Total de %s: %d" % (animais[i],count[i]))
for i in range(len(animais)):
    percent = ((count[i]*100)/TOTAL)
    print("Percentual de %s: %.2f %%" % (animais[i], percent))

File: C++/1094 - Experiments.cpp
#include <iostream>
#include <vector>
#include <cstdio>
 
void print_results(std::vector <int> nums)
{
    std::vector <std::string> animals = {"coelhos","ratos","sapos"};
    printf("Total: %d cobaias\n", nums[3]);
    for (int i = 0; i < 3; i++)
    {
        printf("Total de %s: %d\n",(animals[i].c_str()),nums[i]);
    }
    for (int i = 0; i < 3; i++)
    {
        double percent;
        percent = (nums[i]/double(nums[3]))*100;
        printf("Percentual de %s: %.2f %%\n",(animals[i].c_str()),percent);
    }
}

std::vector<int> get_data(int n)
{
    int x;
    char c;
    std::vector <int> nums = {0,0,0,0};
    for(int i = 0; i < n; i++)
    {
        std::cin >> x >> c;
        switch (c)
        {
        case 'C':
            nums[0] +=  x;
            nums[3] += x;
            break;
        case 'R':
            nums[1] += x;
            nums[3] += x;
            break;
        case 'S':
            nums[2] += x;
            nums[3] += x;
            break;
        default:
            break;
        }
    }
    return nums;
}

int main() {
    int n;
    std::cin >> n;
    std::vector<int> nums = get_data(n);
    print_results(nums);
    return 0;
}